ledger/ledgercore/statedelta.go

```diff
@@ -386,47 +391,6 @@ func (ad *AccountDeltas) UpsertAssetResource(addr basics.Address, aidx basics.As
 	ad.assetResourcesCache[key] = last
 }
 
-// OptimizeAllocatedMemory by reallocating maps to needed capacity
-// For each data structure, reallocate if it would save us at least 50MB aggregate
-func (sd *StateDelta) OptimizeAllocatedMemory(proto config.ConsensusParams) {
-	// accts takes up 232 bytes per entry, and is saved for 320 rounds
-	if uint64(cap(sd.Accts.accts)-len(sd.Accts.accts))*accountArrayEntrySize*proto.MaxBalLookback > stateDeltaTargetOptimizationThreshold {
-		accts := make([]NewBalanceRecord, len(sd.Accts.acctsCache))
-		copy(accts, sd.Accts.accts)
-		sd.Accts.accts = accts
-	}
-
-	// acctsCache takes up 64 bytes per entry, and is saved for 320 rounds
-	// realloc if original allocation capacity greater than length of data, and space difference is significant
-	if 2*sd.initialTransactionsCount > len(sd.Accts.acctsCache) &&
-		uint64(2*sd.initialTransactionsCount-len(sd.Accts.acctsCache))*accountMapCacheEntrySize*proto.MaxBalLookback > stateDeltaTargetOptimizationThreshold {
-		acctsCache := make(map[basics.Address]int, len(sd.Accts.acctsCache))
-		for k, v := range sd.Accts.acctsCache {
-			acctsCache[k] = v
-		}
-		sd.Accts.acctsCache = acctsCache
-	}
-
-	// TxLeases takes up 112 bytes per entry, and is saved for 1000 rounds
-	if sd.initialTransactionsCount > len(sd.Txleases) &&
-		uint64(sd.initialTransactionsCount-len(sd.Txleases))*txleasesEntrySize*proto.MaxTxnLife > stateDeltaTargetOptimizationThreshold {
-		txLeases := make(map[Txlease]basics.Round, len(sd.Txleases))
```

Comment:
we just saw that removing this resulted in 2.5GB memory usage, are these and other optimizations worth bringing back?